search/collector/topn.go

```diff
@@ -39,6 +39,9 @@ type collectorStore interface {
 	AddNotExceedingSize(doc *search.DocumentMatch, size int) *search.DocumentMatch
 
 	Final(skip int, fixup collectorFixup) (search.DocumentMatchCollection, error)
+
+	// Provide access the internal heap implementation
+	Internal() search.DocumentMatchCollection
```

Comment:
`collectStoreList` needs to deploy this as well?

Reply:
collectStoreHeap and collectStoreHeap are both impl of the collector interface and represent a general-purpose K-size heap. collectStoreList just skips usage of golang's built-in heap since its faster for K < 10. So hence both the collectStoreList and collectStoreHeap should impl the Internal() api that just provides access to the internal slice being used for the heap.

Reply:
Right, but I don't see where you're implementing the Internal() api for collectStoreList.

Reply:
oh ok i got confused between collectStoreSlice and collectStoreList 
i see that only collectStoreSlice and collectStoreHeap is used and collectStoreList is unused / deprecated. Hence i had forgot to implement for collectStoreList.